test/inspect-provenance.test.ts

```diff
@@ -0,0 +1,89 @@
+/* eslint-disable @typescript-eslint/no-non-null-assertion */
+import { test } from 'tap';
+import { chdirWorkspaces, activateVirtualenv } from './test-utils';
+
+import pluginImpl = require('../lib');
+
+const pipAppExpectedDependenciesOnlyProvenance = {
+  django: {
+    data: {
+      name: 'django',
+      version: '1.6.1',
+      labels: { provenance: 'requirements.txt:2' },
+    },
+    msg: 'django looks ok',
+  },
+  jinja2: {
+    data: {
+      name: 'jinja2',
+      version: '2.7.2',
+      labels: { provenance: 'requirements.txt:1' },
+    },
+    msg: 'jinja2 looks ok',
+  },
+  'python-etcd': {
+    data: {
+      name: 'python-etcd',
+      version: '0.4.5',
+      labels: { provenance: 'requirements.txt:3' },
+    },
+    msg: 'python-etcd is ok',
+  },
+  'django-select2': {
+    data: {
+      name: 'django-select2',
+      version: '6.0.1',
+      labels: { provenance: 'requirements.txt:4' },
+    },
+    msg: 'django-select2 looks ok',
+  },
+  irc: {
+    data: {
+      name: 'irc',
+      version: '16.2',
+      labels: { provenance: 'requirements.txt:5' },
+    },
+    msg: 'irc ok, even though it has a cyclic dep, yay!',
+  },
+  testtools: {
+    data: {
+      name: 'testtools',
+      version: '2.3.0',
+      labels: { provenance: 'requirements.txt:6' },
+    },
+    msg: "testtools ok, even though it's cyclic, yay!",
+  },
+};
+
+test('inspect --only-provenance', async (t) => {
+  chdirWorkspaces('pip-app');
+  t.teardown(activateVirtualenv('pip-app'));
+  const result = await pluginImpl.inspect('.', 'requirements.txt', {
```

Comment:
how about a test for what happens when this is called with Pipfile?

Reply:
Added.